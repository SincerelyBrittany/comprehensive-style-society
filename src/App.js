import './App.scss';
import './Filter.scss';
import ExportCode from './components/ExportCode';
import CustomColorPicker from './components/ColorPicker'
import React, { Component } from 'react';

export default class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      fontSize: 16,
      fontFamily: "",
      fontEmphasis: 500,
      width: 150,
      height: 100,
      padding: 3,
      borderWidth: 1,
      borderRadius: 0,
      shadow: 0,
      backgroundColor: "",
      textColor: "",
      borderColor: "",
      opacity: 0,
      showCode: false

    }
    this.showCode = this.showCode.bind(this);
  }
  showCode() {
    this.setState(state => ({
        showCode: !state.showCode
    }));
  }
  
  /* functions to change button styling */
  fontSlider(e) {
    this.setState({fontSize: e.target.value});
    document.getElementById('generatedButton').style.fontSize = e.target.value + 'px';
  }
  fontSizeSlider(e) {
    this.setState({fontSize: e.target.value});
    document.getElementById('generatedButton').style.fontSize = e.target.value + 'px';
  }
  fontEmphasisSlider(e) {
    this.setState({fontEmphasis: e.target.value});
    document.getElementById('generatedButton').style.fontWeight = e.target.value;
  }
  widthSlider(e) {
    this.setState({width: e.target.value});
    document.getElementById('generatedButton').style.width = e.target.value + 'px';
  }
  heightSlider(e) {
    this.setState({height: e.target.value});
    document.getElementById('generatedButton').style.height = e.target.value + 'px';
  }
  paddingSlider(e) {
    this.setState({padding: e.target.value});
    document.getElementById('generatedButton').style.padding = e.target.value + 'px';
  }
  borderWidthSlider(e) {
    this.setState({borderWidth: e.target.value});
    document.getElementById('generatedButton').style.borderWidth = e.target.value + 'px';
  }
  borderRadiusSlider(e) {
    this.setState({borderRadius: e.target.value});
    document.getElementById('generatedButton').style.borderRadius = e.target.value + 'px';
  }
  borderShadowSlider(e) {
    this.setState({shadow: e.target.value});
    document.getElementById('generatedButton').style.boxShadow = '0 0px ' + e.target.value + 'px ' + e.target.value + 'px rgba(0,0,0,0.2)';
  }
  opacitySlider(e) {
    this.setState({opacity: e.target.value});
    document.getElementById('generatedButton').style.opacity = e.target.value + '%';
  }

  render() {
    return (
      <div className="App">
        <div className="column left-column">
          <div className="presets card">
            <p>Basic Presets</p>
            <div className="preset-container">
              <div className="preset-row">
                <div className="preset">
                  <button id="normal-effect">Button</button>
                </div>
                <div className="preset">
                  <button id="opacity-effect">Button</button>
                </div>
              </div>
              <div className="preset-row">
                <div className="preset">
                  <button id="rounded-corners1-effect">Button</button>
                </div>
                <div className="preset">
                  <button id="rounded-corners2-effect">Button</button>
                </div>
              </div>
            </div>
          </div>
          <div className="presets card">
            <p>Other Presets</p>
            <div className="preset-container">
              <div className="preset-row">
                <div className="preset">
                  <button id="thin-border-effect">Button</button>
                </div>
                <div className="preset">
                  <button id="thick-border-effect">Button</button>
                </div>
              </div>
              <div className="preset-row">
                <div className="preset">
                  <button id="hover-normal-fill-effect">Button</button>
                </div>
                <div className="preset">
                  <button id="hover-opaque-fill-effect">Button</button>
                </div>
              </div>
              <div className="preset-row">
                <div className="preset">
                  <button id="shadow-effect">Button</button>
                </div>
                <div className="preset">
                  <button id="hover-shadow-effect">Button</button>
                </div>
              </div>
              <div className="preset-row">
                <div className="preset">
                  <button id="hover-grow-effect">Button</button>
                </div>
                <div className="preset">
                  <button id="hover-shrink-effect">Button</button>
                </div>
              </div>
              <div className="preset-row">
                <div className="preset">
                  <button id="hover-slide-right-effect">Button</button>
                </div>
                <div className="preset">
                  <button id="hover-slide-left-effect">Button</button>
                </div>
              </div>
              <div className="preset-row">
                <div className="preset">
                  <button id="hover-rotate-right-effect">Button</button>
                </div>
                <div className="preset">
                  <button id="hover-rotate-left-effect">Button</button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="column center-column">
          <div className="title">
            CSS Generator
          </div>
          <div className="button-window card">
            <button id="generatedButton">
              Button
            </button>
          </div>
          <div className="json-window card">
            <button onClick={this.showCode}>
                Show CSS
            </button>
            {this.state.showCode ? <ExportCode fontSize={this.state.fontSize} 
            fontFamily = {this.state.fontFamily}
            fontEmphasis = {this.state.fontEmphasis}
            width = {this.state.width}
            height = {this.state.height}
            padding = {this.state.padding}
            borderWidth = {this.state.borderWidth}
            borderRadius = {this.state.borderRadius}
            shadow = {this.state.shadow}
            textColor = {this.state.textColor}
            borderColor = {this.state.borderColor}
            opacity = {this.state.opacity}
            backgroundColor = {this.state.backgroundColor} false/> : null }
          </div>
        </div>
        <div className="column right-column">
          <div className="filters three card">
            <p className="filterType">Font</p>
            <div className="filters-container">
              <div className="attribute">
                <p>Font Family</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.fontSize} onChange={(e) => this.fontSlider(e)} min="0" max="100"/>
                  <span className="range-slider__value">{this.state.fontSize}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Font Weight</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.fontEmphasis} onChange={(e) => this.fontEmphasisSlider(e)} min="0" max="900" step="100"/>
                  <span className="range-slider__value">{this.state.fontEmphasis}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Font Size</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.fontSize} onChange={(e) => this.fontSizeSlider(e)} min="0" max="100"/>
                  <span className="range-slider__value">{this.state.fontSize}</span>
                </div>
              </div>
            </div>
          </div>
          <div className="filters three card">
            <p className="filterType">Sizing</p>
            <div className="filters-container">
              <div className="attribute">
                <p>Width</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.width} onChange={(e) => this.widthSlider(e)} min="0" max="500"/>
                  <span className="range-slider__value">{this.state.width}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Height</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.height} onChange={(e) => this.heightSlider(e)} min="0" max="175"/>
                  <span className="range-slider__value">{this.state.height}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Padding</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.padding} onChange={(e) => this.paddingSlider(e)} min="0" max="50"/>
                  <span className="range-slider__value">{this.state.padding}</span>
                </div>
              </div>
            </div>
          </div>
          <div className="filters three card">
            <p className="filterType">Border</p>
            <div className="filters-container">
              <div className="attribute">
                <p>Border Width</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.borderWidth} onChange={(e) => this.borderWidthSlider(e)} min="0" max="10"/>
                  <span className="range-slider__value">{this.state.borderWidth}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Border Radius</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.borderRadius} onChange={(e) => this.borderRadiusSlider(e)} min="0" max="15"/>
                  <span className="range-slider__value">{this.state.borderRadius}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Shadow</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.shadow} onChange={(e) => this.borderShadowSlider(e)} min="0" max="100"/>
                  <span className="range-slider__value">{this.state.shadow}</span>
                </div>
              </div>              
            </div>
          </div>
          <div className="filters four card">
            <p className="filterType">Color</p>
            <div className="filters-container">
              <div className="attribute">
                <p>Background Color</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.fontSize} onChange={(e) => this.fontSlider(e)} min="0" max="100"/>
                  <span className="range-slider__value">{this.state.fontSize}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Text Color</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.fontSize} onChange={(e) => this.fontSlider(e)} min="0" max="100"/>
                  <span className="range-slider__value">{this.state.fontSize}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Border Color</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.fontSize} onChange={(e) => this.fontSlider(e)} min="0" max="100"/>
                  <span className="range-slider__value">{this.state.fontSize}</span>
                </div>
              </div>
              <div className="attribute">
                <p>Opacity</p>
                <div className="range-slider">
                  <input className="range-slider__range" type="range" value={this.state.opacity} onChange={(e) => this.opacitySlider(e)} min="0" max="100"/>
                  <span className="range-slider__value">{this.state.opacity}</span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
  
}
