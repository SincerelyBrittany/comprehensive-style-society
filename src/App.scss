// color schemes : https://graphicmama.com/blog/website-color-schemes/
@import url('https://fonts.googleapis.com/css2?family=Montserrat&display=swap');

.App {
  text-align: center;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: #f6f6f2;
  height: 100vh;
  width: 100vw;
  font-family: 'Montserrat', sans-serif;
}

.title {
  text-align: center;
  font-size: 70px;
  color: #388087
}

.showButton {
  display:inline-block;
  padding:0.7em 1.4em;
  margin:0 0.3em 0.3em 0;
  box-sizing: border-box;
  text-decoration:none;
  font-family:'montserrat',sans-serif;
  text-transform:uppercase;
  font-weight:400;
  color:#FFFFFF;
  background-color:#3369ff;
  box-shadow:inset 0 -0.6em 0 -0.35em rgba(0,0,0,0.17);
  text-align:center;
  position:relative;
  background-color: #388087;
  border: 0;
}

.highlight {
  height: 5px;
  width: 100%;
  background-color: #c2edce;
  border-radius: 5px 5px 0 0;
}

.highlight2 {
  height: 5px;
  width: 100%;
  background-color: #c2edce;
  border-radius: 5px 5px 0 0;
}
//badfe7

.column {
  overflow: auto;
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 10px;
}

.center-column {
  width: 39%;
}

.left-column {
  width: 29%;
  overflow: auto;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.right-column {
  width: 29%;
  overflow: auto;
  height: 100vh;
}

.card {
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  border-radius: 5px;
  background-color: white;
  margin: 10px;
}

.button-window {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 30vh;
  width: 100%
}

.json-window {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
  width: 100%;
  height: 40vh;
}

.presets {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 25vh;
  width: 100%
}

.preset-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.preset-row {
  display: flex; 
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 50%;
}

.preset {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  border: 0px solid black;
  width: 40%;
  height: 80%;
  margin: 8px;
}

.filters {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 32vh;
  width: 100%
}

.two {
  height: 38vh;
}

.four {
  height: 59vh;
}

.three {
  height: 48vh;
}

.six {
  height: 60vh;
}

.filters-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.slider {
  background: lightgrey;
  outline: none;
}

.attribute {
  display: flex;
  flex-direction: column;
  text-align: center;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.filterType {
  font-weight: 700;
  margin-bottom: 0;
}

#normal-effect {
  background-color: #6fb3b8;
  color: white;
  font-size: 16px;
  cursor: pointer;
}

#opacity-effect {
  background-color: #6fb3b8;
  opacity: 0.25;
  color: white;
  font-size: 16px;
  cursor: pointer;
}

#rounded-corners1-effect {
  background-color: #6fb3b8;
  border-radius: 10px;
  color: white;
  font-size: 16px;
  cursor: pointer;
}

#rounded-corners2-effect {
  background-color: #6fb3b8;
  border-radius: 50%;
  color: white;
  font-size: 16px;
  cursor: pointer;
}

#thin-border-effect {
  background-color: #F6F6F6;
  border: 2px solid #6fb3b8;
  border-radius: 10px;
  color: black;
  font-size: 16px;
  cursor: pointer;
}

#thick-border-effect {
  background-color: #F6F6F6;
  border: 6px solid #6fb3b8;
  border-radius: 10px;
  color: black;
  font-size: 16px;
  cursor: pointer;
}

#hover-normal-fill-effect {
  background-color: #F6F6F6;
  border: 5px solid #6fb3b8;
  border-radius: 10px;
  color: black;
  font-size: 16px;
  transition-duration: 0.5s;
  cursor: pointer;
}

#hover-normal-fill-effect:hover {
  background-color: #6fb3b8;
  color: white;
}

#hover-opaque-fill-effect {
  background-color: #F6F6F6;
  border: 5px solid #6fb3b8;
  border-radius: 10px;
  color: black;
  font-size: 16px;
  transition-duration: 0.5s;
  cursor: pointer;
}

#hover-opaque-fill-effect:hover {
  background-color: #6fb3b8;
  opacity: 0.25;
  color: white;
}

#shadow-effect {
  background-color: #6fb3b8;
  color: white;
  font-size: 16px;
  cursor: pointer;
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19);
}

#hover-shadow-effect {
  background-color: #6fb3b8;
  color: white;
  font-size: 16px;
  cursor: pointer;
}

#hover-shadow-effect:hover {
  box-shadow: 0 16px 24px 0 rgba(0,0,0,0.28), 0 17px 50px 0 rgba(0,0,0,0.19);
}

#hover-grow-effect {
  background-color: #6fb3b8;
  color: white;
  font-size: 16px;
  cursor: pointer;
  transition: all 0.5s ease-in-out;
}

#hover-grow-effect:hover {
  transform: scale(1.25);
}

#hover-shrink-effect {
  background-color: #6fb3b8;
  color: white;
  font-size: 16px;
  cursor: pointer;
  transition: all 0.5s ease-in-out;
}

#hover-shrink-effect:hover {
  transform: scale(0.75);
}

#hover-slide-right-effect {
  background-color: #F6F6F6;
  border: 2px solid #6fb3b8;
  color: black;
  font-size: 16px;
  cursor: pointer;
  background: linear-gradient(to left, #F6F6F6 50%, #6fb3b8 50%) right;
  background-size: 200%;
  transition: 0.5s ease-out;
}

#hover-slide-right-effect:hover {
  background-position: left;
  color: white;
}

#hover-slide-left-effect {
  background-color: #F6F6F6;
  border: 2px solid #6fb3b8;
  color: black;
  font-size: 16px;
  cursor: pointer;
  background: linear-gradient(to right, #F6F6F6 50%, #6fb3b8 50%) left;
  background-size: 200%;
  transition: 0.5s ease-out;
}

#hover-slide-left-effect:hover {
  background-position: right;
  color: white;
}

#hover-rotate-right-effect {
  background-color: #6fb3b8
  color: white;
  font-size: 16px;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

#hover-rotate-right-effect:hover {
  transform: scale(1.1) rotate(6deg);
}

#hover-rotate-left-effect {
  background-color: #6fb3b8;
  color: white;
  font-size: 16px;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

#hover-rotate-left-effect:hover {
  transform: scale(1.1) rotate(-6deg);
}