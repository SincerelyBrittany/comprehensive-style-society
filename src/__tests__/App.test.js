import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../App';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
})

// it('changes the state for font slider', () => {
//     const component = renderer.create(<App/>);
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
  
//     // manually trigger the callback
//     tree.state.fontSlider();
//     // re-rendering
//     tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
  
//     // manually trigger the callback
//     tree.props.onMouseLeave();
//     // re-rendering
//     tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });