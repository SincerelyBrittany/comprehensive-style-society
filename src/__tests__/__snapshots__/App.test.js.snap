// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the state for font slider 1`] = `
<div
  className="App"
>
  <div
    className="column left-column"
  >
    <div
      className="presets card"
    >
      <p>
        Basic Presets
      </p>
      <div
        className="preset-container"
      >
        <div
          className="preset-row"
        >
          <div
            className="preset"
          >
            <button
              id="normal-effect"
            >
              Button
            </button>
          </div>
          <div
            className="preset"
          >
            <button
              id="opacity-effect"
            >
              Button
            </button>
          </div>
        </div>
        <div
          className="preset-row"
        >
          <div
            className="preset"
          >
            <button
              id="rounded-corners1-effect"
            >
              Button
            </button>
          </div>
          <div
            className="preset"
          >
            <button
              id="rounded-corners2-effect"
            >
              Button
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="presets card"
    >
      <p>
        Other Presets
      </p>
      <div
        className="preset-container"
      >
        <div
          className="preset-row"
        >
          <div
            className="preset"
          >
            <button
              id="thin-border-effect"
            >
              Button
            </button>
          </div>
          <div
            className="preset"
          >
            <button
              id="thick-border-effect"
            >
              Button
            </button>
          </div>
        </div>
        <div
          className="preset-row"
        >
          <div
            className="preset"
          >
            <button
              id="hover-normal-fill-effect"
            >
              Button
            </button>
          </div>
          <div
            className="preset"
          >
            <button
              id="hover-opaque-fill-effect"
            >
              Button
            </button>
          </div>
        </div>
        <div
          className="preset-row"
        >
          <div
            className="preset"
          >
            <button
              id="shadow-effect"
            >
              Button
            </button>
          </div>
          <div
            className="preset"
          >
            <button
              id="hover-shadow-effect"
            >
              Button
            </button>
          </div>
        </div>
        <div
          className="preset-row"
        >
          <div
            className="preset"
          >
            <button
              id="hover-grow-effect"
            >
              Button
            </button>
          </div>
          <div
            className="preset"
          >
            <button
              id="hover-shrink-effect"
            >
              Button
            </button>
          </div>
        </div>
        <div
          className="preset-row"
        >
          <div
            className="preset"
          >
            <button
              id="hover-slide-right-effect"
            >
              Button
            </button>
          </div>
          <div
            className="preset"
          >
            <button
              id="hover-slide-left-effect"
            >
              Button
            </button>
          </div>
        </div>
        <div
          className="preset-row"
        >
          <div
            className="preset"
          >
            <button
              id="hover-rotate-right-effect"
            >
              Button
            </button>
          </div>
          <div
            className="preset"
          >
            <button
              id="hover-rotate-left-effect"
            >
              Button
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="column center-column"
  >
    <div
      className="title"
    >
      CSS Generator
    </div>
    <div
      className="button-window card"
    >
      <button
        id="generatedButton"
      >
        Button
      </button>
    </div>
    <div
      className="json-window card"
    >
      <button
        onClick={[Function]}
      >
        Show CSS
      </button>
    </div>
  </div>
  <div
    className="column right-column"
  >
    <div
      className="filters three card"
    >
      <p
        className="filterType"
      >
        Font
      </p>
      <div
        className="filters-container"
      >
        <div
          className="attribute"
        >
          <p>
            Font Family
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="100"
              min="0"
              onChange={[Function]}
              type="range"
              value={16}
            />
            <span
              className="range-slider__value"
            >
              16
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Font Weight
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="900"
              min="0"
              onChange={[Function]}
              step="100"
              type="range"
              value={500}
            />
            <span
              className="range-slider__value"
            >
              500
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Font Size
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="100"
              min="0"
              onChange={[Function]}
              type="range"
              value={16}
            />
            <span
              className="range-slider__value"
            >
              16
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="filters three card"
    >
      <p
        className="filterType"
      >
        Sizing
      </p>
      <div
        className="filters-container"
      >
        <div
          className="attribute"
        >
          <p>
            Width
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="500"
              min="0"
              onChange={[Function]}
              type="range"
              value={150}
            />
            <span
              className="range-slider__value"
            >
              150
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Height
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="175"
              min="0"
              onChange={[Function]}
              type="range"
              value={100}
            />
            <span
              className="range-slider__value"
            >
              100
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Padding
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="50"
              min="0"
              onChange={[Function]}
              type="range"
              value={3}
            />
            <span
              className="range-slider__value"
            >
              3
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="filters three card"
    >
      <p
        className="filterType"
      >
        Border
      </p>
      <div
        className="filters-container"
      >
        <div
          className="attribute"
        >
          <p>
            Border Width
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="10"
              min="0"
              onChange={[Function]}
              type="range"
              value={1}
            />
            <span
              className="range-slider__value"
            >
              1
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Border Radius
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="15"
              min="0"
              onChange={[Function]}
              type="range"
              value={0}
            />
            <span
              className="range-slider__value"
            >
              0
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Shadow
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="100"
              min="0"
              onChange={[Function]}
              type="range"
              value={0}
            />
            <span
              className="range-slider__value"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="filters four card"
    >
      <p
        className="filterType"
      >
        Color
      </p>
      <div
        className="filters-container"
      >
        <div
          className="attribute"
        >
          <p>
            Background Color
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="100"
              min="0"
              onChange={[Function]}
              type="range"
              value={16}
            />
            <span
              className="range-slider__value"
            >
              16
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Text Color
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="100"
              min="0"
              onChange={[Function]}
              type="range"
              value={16}
            />
            <span
              className="range-slider__value"
            >
              16
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Border Color
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="100"
              min="0"
              onChange={[Function]}
              type="range"
              value={16}
            />
            <span
              className="range-slider__value"
            >
              16
            </span>
          </div>
        </div>
        <div
          className="attribute"
        >
          <p>
            Opacity
          </p>
          <div
            className="range-slider"
          >
            <input
              className="range-slider__range"
              max="100"
              min="0"
              onChange={[Function]}
              type="range"
              value={0}
            />
            <span
              className="range-slider__value"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
